app/src/main/java/org/mozilla/fenix/compose/tabstray/TabGridItem.kt

```diff
@@ -0,0 +1,336 @@
+ */
+@Composable
+private fun FadingEndEdge(
+    modifier: Modifier = Modifier,
+    fadeWidth: Dp = 25.dp,
+    backgroundColor: Color = Color.Transparent,
+    isContentRtl: Boolean = false,
+    content: @Composable () -> Unit
+) {
+    // List of colors defining the direction of the fade effect
+    val colorList = listOf(Color.Transparent, backgroundColor)
+    CompositionLocalProvider(LocalLayoutDirection provides LayoutDirection.Ltr) {
+        Box(modifier) {
+            content()
+            Spacer(
+                Modifier
+                    .width(fadeWidth)
+                    .fillMaxHeight()
+                    .align(
+                        if (isContentRtl) {
+                            Alignment.CenterStart
+                        } else {
+                            Alignment.CenterEnd
+                        }
+                    )
+                    .background(
+                        Brush.horizontalGradient(
+                            colors = if (isContentRtl) {
+                                colorList.reversed()
+                            } else {
+                                colorList
+                            }
+                        )
+                    )
+            )
+        }
+    }
+}
+
+/**
+ * Thumbnail specific for the [TabGridItem], which can be selected.
+ *
+ * @param tab Tab, containing the thumbnail to be displayed.
+ * @param multiSelectionSelected Whether or not the multiple selection is enabled.
+ */
+@Composable
+private fun Thumbnail(
+    tab: TabSessionState,
+    multiSelectionSelected: Boolean,
+) {
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(FirefoxTheme.colors.layer2)
+    ) {
+        TabThumbnail(
+            tabId = tab.id,
+            size = LocalConfiguration.current.screenWidthDp.dp
+        )
+        if (multiSelectionSelected) {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .background(FirefoxTheme.colors.layerAccentNonOpaque)
+            )
+            Card(
+                modifier = Modifier
+                    .size(size = 40.dp)
+                    .align(alignment = Alignment.Center),
+                shape = CircleShape,
+                backgroundColor = FirefoxTheme.colors.layerAccent,
+            ) {
+                Icon(
+                    painter = painterResource(id = R.drawable.mozac_ic_check),
+                    modifier = Modifier
+                        .matchParentSize()
+                        .padding(all = 8.dp),
+                    contentDescription = null,
+                    tint = colorResource(id = R.color.mozac_ui_icons_fill)
+                )
+            }
+        }
+    }
+}
+
+@Composable
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES)
+@Preview(uiMode = Configuration.UI_MODE_NIGHT_NO)
+private fun TabGridItemPreview() {
+    FirefoxTheme(theme = Theme.getTheme()) {
```

Comment:
nit: the theme no longer needs to get set in order to get previews to build (the default value is what's being passed-in here)

```suggestion
    FirefoxTheme {
```